src/core/value.js

```
@@ -0,0 +1,134 @@
+    const val = this._value;
+    const mult = multiplicand._value;
+    const result = (val).times(mult);
+    return this._canonicalize(result);
+  }
+
+  scale(scaleFactor: Value) {
+    const result = this._value.times(scaleFactor._value);
+    return this._canonicalize(result);
+
+  }
+
+  divide(divisor: Value) {
+    if (this._value.isNaN()) {
+      throw new Error('Invalid dividend');
+    }
+    if (divisor.isNaN()) {
+      throw new Error('Invalid divisor');
+    }
+    if (divisor.isZero()) {
+      throw new Error('divide by zero');
+    }
+    const result = this._value.dividedBy(divisor._value);
+    return this._canonicalize(result);
+  }
+
+  invert() {
+    const result = (new BigNumber(this._value)).toPower(-1);
+    return this._canonicalize(result);
+  }
+
+  round(decimalPlaces, roundingMode) {
+    this._value = this._value.round(decimalPlaces, roundingMode);
+    return this;
+  }
+
+  toPrecision(significantDigits, roundingMode) {
+    this._value =
+      new BigNumber(this._value.toPrecision(significantDigits, roundingMode));
+    return this;
+  }
+
+  toFixed(decimalPlaces, roundingMode) {
+    return this._value.toFixed(decimalPlaces, roundingMode);
+  }
+
+  getExponent() {
+    return this._value.e;
+  }
+
+  getMantissaDecimalString() {
+    let mantissa = this._value.abs().toPrecision(16)
+    .replace(/\./, '')      // remove decimal point
+    .replace(/e.*/, '')     // remove scientific notation
+    .replace(/^0*/, '');    // remove leading zeroes
+    while (mantissa.length < 16) {
+      mantissa += '0';        // add trailing zeroes until length is 16
+    }
+    return mantissa;
+  }
+
+  isNaN() {
+    return this._value.isNaN();
+  }
+
+  isZero() {
+    return this._value.isZero();
+  }
+
+  isNegative() {
+    return this._value.isNegative();
+  }
+
+  toString() {
+    return this._value.toString();
+  }
+
+  greaterThan(comparator: Value) {
+    return this._value.greaterThan(comparator._value);
```

Comparisons should probably throw an exception if you try to compare different subtypes.